packages/xo-server/src/api/vm.js

```diff
@@ -740,22 +741,31 @@ export { convertToTemplate as convert }
 // -------------------------------------------------------------------
 
 // TODO: implement resource sets
-export async function snapshot ({
-  vm,
-  name = `${vm.name_label}_${new Date().toISOString()}`,
-}) {
+export const snapshot = defer(async function (
+  $defer,
+  { vm, name = `${vm.name_label}_${new Date().toISOString()}` }
+) {
   await checkPermissionOnSrs.call(this, vm)
 
-  return (await this.getXapi(vm).snapshotVm(vm._xapiRef, name)).$id
-}
+  const xapi = this.getXapi(vm)
+  const snapshotId = (await xapi.snapshotVm(vm._xapiRef, name)).$id
```

Comment:
You could use destructuring here, it goes neatly with `await` :slightly_smiling_face: